roles/sos_report/tasks/sos-reports.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: Generate SOS report for node {{ node_name }}
+  vars:
+    sos_report_registry: "{{ sos_report_image | regex_replace('([^/]+)/.*', '\\1') }}"
+    sos_report_image_name: "{{ sos_report_image | regex_replace('[^/]+/(.*)', '\\1') }}"
+  shell: >
+    {{ sos_report_oc_path }} debug --image={{ sos_report_image }} node/{{ node_name }} --
+    bash -c 'echo -e REGISTRY='{{ sos_report_registry }}'\\nIMAGE='{{ sos_report_image_name }}'> /host/root/.toolboxrc';
+    {{ sos_report_oc_path }} debug --image={{ sos_report_image }} node/{{ node_name }} --
+    chroot /host
+    toolbox
+    sos report --batch {{ sos_report_options }}
+  async: 600
+  poll: 0
+  register: report
+  loop: "{{ sos_report_nodes }}"
+  loop_control:
+    loop_var: node_name
+  changed_when: true
+
+- name: Check SOS report status
+  async_status:
+    jid: "{{ result.ansible_job_id }}"
+  loop: "{{ report.results }}"
+  loop_control:
+    loop_var: "result"
+  register: sos_async_results
+  until: sos_async_results.finished
+  retries: 20
+  delay: 30
+
+- name: Extract SOS report for node {{ async.result.node_name }}
+  vars:
+    tarball: "{{ async.stdout | regex_findall('/host/var/tmp/sosreport-.*.tar.xz') | first }}"
```

Comment:
if you're finding the first tarball found, you have to clean the tarballs at some point either before you run the sos report (preferred) or after you've processed the tarball

Reply:
> if you're grabbing the first tarball found, 

This `first` is to extract the unique element out of the list, see this is a loop, so it does on every iteration.

> you have to clean the tarballs at some point either before you run the sos report (preferred) or after you've processed the tarball

Not necessarily, the tarballs have a timestamp in the name that makes them unique.

Reply:
what's the current behavior if you run sos_report through this module multiple times? is it safe? how big are ,SOS report files?

Reply:
> what's the current behavior if you run sos_report through this module multiple times? is it safe? how big are SOS report files?

A tarball is created for each node defined, each tarball is unique to the run. 

Don't understand the relevance of the size of the tarballs as that is not  under the control of this role.

Reply:
It is maybe impact the storage of the DCI infrastructure :D

Reply:
> It is maybe impact the storage of the DCI infrastructure :D

Not for this role, though, since the role does not upload the sos reports to DCI, that would be a concern for the agent that picks these tarballs into DCI :smile:

Reply:
I'm not trying to be difficult, I was just directing attention to the possible failure scenarios on this role, if there are no failure scenarios then cool by me.